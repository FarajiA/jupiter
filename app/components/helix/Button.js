import React from 'react';
import PropTypes from 'prop-types';

class Button extends React.Component {
  render() {
    const { label, disabled, classNames, type, onClick, children } = this.props;
    return (
      <button className={`hxBtn ${classNames}`} type={type} disabled={disabled} onClick={onClick}>
        {label}
        {children}
      </button>
    );
  }
}

Button.propTypes = {
  children: PropTypes.node,
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  classNames: PropTypes.string,
  type: PropTypes.string
};

Button.defaultProps = {
  disabled: false,
  classNames: '',
  type: 'button'
};


export default Button;
