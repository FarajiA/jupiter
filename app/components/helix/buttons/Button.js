import React from 'react';
import PropTypes from 'prop-types';

class Button extends React.Component {
  render() {
    const { label, disabled, classNames, onClick, processing, submit, children } = this.props;
    return (
      <button // eslint-disable-line react/button-has-type
        className={`hxBtn ${classNames}`}
        type={submit ? 'submit' : 'button'}
        disabled={disabled}
        onClick={onClick}
      >
        {!processing && (label || children)}
        {processing && <span>{label || children}</span>}
        {processing && <hx-busy />}
      </button>
    );
  }
}

Button.propTypes = {
  children: PropTypes.node,
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  classNames: PropTypes.string,
  processing: PropTypes.bool,
  submit: PropTypes.bool
};

Button.defaultProps = {
  disabled: false,
  classNames: '',
  submit: false
};


export default Button;
