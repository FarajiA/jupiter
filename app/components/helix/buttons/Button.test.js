import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('Button', () => {
  const clickMock = jest.fn();
  const props = {
    label: 'Click Me!',
    variant: 'primary',
    wide: true,
    classNames: 'new-class',
    processing: false,
    onClick: clickMock
  };
  test('it renders correct props', () => {
    const component = renderer.create(<Button {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Label prop is rendered', () => {
    const wrapper = mount(<Button {...props} />);
    expect(wrapper.text()).toEqual('Click Me!');
  });

  test('correct type and classes are rendered if submit', () => {
    const wrapper = shallow(<Button {...props} submit />);
    expect(wrapper.prop('className')).toContain('submit-btn');
    expect(wrapper.prop('type')).toEqual('submit');
  });

  test('busy icon is rendered if processing', () => {
    const icon = mount(<Button {...props} processing />).find('hx-busy');
    expect(icon.length).toBeTruthy();
  });

  test('onClick is invoked upon click', () => {
    const wrapper = shallow(<Button {...props} />);
    expect(clickMock).toHaveBeenCalledTimes(0);
    wrapper.simulate('click');
    expect(clickMock).toHaveBeenCalled();
  });
});
