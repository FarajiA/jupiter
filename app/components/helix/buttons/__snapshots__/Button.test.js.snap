// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button it renders correct props 1`] = `
<button
  className="hxBtn hxPrimary  new-class btn-wide"
  disabled={false}
  onClick={[MockFunction]}
  type="button"
>
  <span>
    Click Me!
  </span>
</button>
`;
